File: archive/old_implementation/backend/schemas/proposal.py
"""
TenderWise AI - Proposal Schemas
"""

from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field
from models.proposal import ProposalStatus


class ProposalDocumentResponse(BaseModel):
    """Proposal document response schema"""
    id: int
    filename: str
    original_filename: str
    file_size: int
    content_type: str
    document_type: Optional[str] = None
    processing_status: str
    created_at: datetime

    class Config:
        from_attributes = True


class ProposalBase(BaseModel):
    """Base proposal schema"""
    title: str = Field(..., min_length=1, max_length=500)
    executive_summary: Optional[str] = None
    technical_approach: Optional[str] = None
    total_cost: Optional[float] = Field(None, ge=0)
    currency: str = Field(default="USD", max_length=3)
    cost_breakdown: Optional[Dict[str, Any]] = None
    proposed_timeline: Optional[Dict[str, Any]] = None
    delivery_date: Optional[datetime] = None
    compliance_matrix: Optional[Dict[str, Any]] = None


class ProposalCreate(ProposalBase):
    """Proposal creation schema"""
    rfp_id: int = Field(..., gt=0)


class ProposalUpdate(ProposalBase):
    """Proposal update schema - all fields optional"""
    title: Optional[str] = Field(None, min_length=1, max_length=500)


class ProposalStatusUpdate(BaseModel):
    """Proposal status update schema"""
    status: ProposalStatus


class ProposalListResponse(BaseModel):
    """Proposal list item response schema"""
    id: int
    rfp_id: int
    title: str
    proposal_number: str
    status: ProposalStatus
    total_cost: Optional[float] = None
    currency: str
    delivery_date: Optional[datetime] = None
    compliance_score: Optional[float] = None
    created_at: datetime
    updated_at: datetime
    submitted_at: Optional[datetime] = None
    
    # Related data
    rfp_title: Optional[str] = None
    rfp_number: Optional[str] = None
    rfp_organization: Optional[str] = None

    class Config:
        from_attributes = True


class ProposalResponse(ProposalBase):
    """Complete proposal response schema"""
    id: int
    rfp_id: int
    proposal_number: str
    status: ProposalStatus
    compliance_score: Optional[float] = None
    evaluation_results: Optional[Dict[str, Any]] = None
    
    # AI Analysis
    strengths: Optional[List[str]] = None
    weaknesses: Optional[List[str]] = None
    risk_factors: Optional[List[str]] = None
    recommendation: Optional[str] = None
    
    # Metadata
    created_by: int
    submitted_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    
    # Related data
    rfp_title: Optional[str] = None
    rfp_number: Optional[str] = None
    rfp_organization: Optional[str] = None
    documents: List[ProposalDocumentResponse] = []

    class Config:
        from_attributes = True


class ProposalEvaluationRequest(BaseModel):
    """Proposal evaluation request schema"""
    evaluate_technical: bool = True
    evaluate_financial: bool = True
    evaluate_compliance: bool = True
    custom_criteria: Optional[Dict[str, float]] = None


class ProposalEvaluationResponse(BaseModel):
    """Proposal evaluation response schema"""
    proposal_id: int
    overall_score: float
    technical_score: Optional[float] = None
    financial_score: Optional[float] = None
    compliance_score: Optional[float] = None
    strengths: List[str] = []
    weaknesses: List[str] = []
    risk_factors: List[str] = []
    recommendation: str
    evaluation_summary: str
    created_at: datetime


class ProposalFilters(BaseModel):
    """Proposal filtering parameters"""
    search: Optional[str] = None
    status: Optional[ProposalStatus] = None
    rfp_id: Optional[int] = None
    my_proposals: bool = False
    min_cost: Optional[float] = None
    max_cost: Optional[float] = None
    order_by: str = "created_at"
    order_direction: str = "desc"

File: frontend/src/types/proposal.ts
export type ProposalStatus = 
  | 'draft' 
  | 'in_progress' 
  | 'under_review' 
  | 'submitted' 
  | 'accepted' 
  | 'rejected' 
  | 'withdrawn'

export interface ProposalDocument {
  id: number
  filename: string
  original_filename: string
  file_size: number
  content_type: string
  document_type?: string
  processing_status: string
  created_at: string
}

export interface ProposalListItem {
  id: number
  rfp_id: number
  title: string
  proposal_number: string
  status: ProposalStatus
  total_cost?: number
  currency: string
  delivery_date?: string
  compliance_score?: number
  created_at: string
  updated_at: string
  submitted_at?: string
  rfp_title?: string
  rfp_number?: string
  rfp_organization?: string
}

export interface Proposal {
  id: number
  rfp_id: number
  title: string
  proposal_number: string
  status: ProposalStatus
  executive_summary?: string
  technical_approach?: string
  total_cost?: number
  currency: string
  cost_breakdown?: Record<string, any>
  proposed_timeline?: Record<string, any>
  delivery_date?: string
  compliance_matrix?: Record<string, any>
  compliance_score?: number
  evaluation_results?: Record<string, any>
  strengths?: string[]
  weaknesses?: string[]
  risk_factors?: string[]
  recommendation?: string
  created_by: number
  submitted_at?: string
  created_at: string
  updated_at: string
  rfp_title?: string
  rfp_number?: string
  rfp_organization?: string
  documents: ProposalDocument[]
}

export interface CreateProposalData {
  rfp_id: number
  title: string
  executive_summary?: string
  technical_approach?: string
  total_cost?: number
  currency?: string
  cost_breakdown?: Record<string, any>
  proposed_timeline?: Record<string, any>
  delivery_date?: string
  compliance_matrix?: Record<string, any>
}

export interface UpdateProposalData {
  title?: string
  executive_summary?: string
  technical_approach?: string
  total_cost?: number
  currency?: string
  cost_breakdown?: Record<string, any>
  proposed_timeline?: Record<string, any>
  delivery_date?: string
  compliance_matrix?: Record<string, any>
}

export interface ProposalEvaluationRequest {
  evaluate_technical: boolean
  evaluate_financial: boolean
  evaluate_compliance: boolean
  custom_criteria?: Record<string, number>
}

export interface ProposalEvaluationResponse {
  proposal_id: number
  overall_score: number
  technical_score?: number
  financial_score?: number
  compliance_score?: number
  strengths: string[]
  weaknesses: string[]
  risk_factors: string[]
  recommendation: string
  evaluation_summary: string
  created_at: string
}

export interface ProposalFilters {
  search?: string
  status?: ProposalStatus
  rfp_id?: number
  my_proposals?: boolean
  min_cost?: number
  max_cost?: number
  order_by?: string
  order_direction?: string
}